ibis/backends/datafusion/__init__.py

```diff
@@ -0,0 +1,141 @@
+        """List the available tables."""
+        tables = list(self.context.tables())
+        if like is not None:
+            pattern = re.compile(like)
+            return list(filter(lambda t: pattern.findall(t), tables))
+        return tables
+
+    def table(self, name, schema=None):
+        catalog = self.context.catalog()
+        database = catalog.database('public')
+        table = database.table(name)
+        schema = sch.infer(table.schema)
+        return self.table_class(name, schema, self).to_expr()
+
+    def register_csv(self, name, path, schema=None):
+        self.context.register_csv(name, path, schema=schema)
+
+    def register_parquet(self, name, path, schema=None):
+        self.context.register_parquet(name, path, schema=schema)
+
+    def execute(
+        self,
+        expr: ir.Expr,
+        params: Mapping[ir.Expr, object] = None,
+        limit: str = 'default',
+        **kwargs,
+    ):
+        def _collect(frame):
+            batches = frame.collect()
+            if batches:
+                table = pa.Table.from_batches(batches)
+            else:
+                # TODO(kszucs): file a bug to datafusion because the fields'
+                # nullability from frame.schema() is not always consistent
+                # with the first record batch's schema
+                table = pa.Table.from_batches(batches, schema=frame.schema())
+            return table.to_pandas()
+
+        if isinstance(expr, ir.TableExpr):
+            frame = self.compile(expr, params, **kwargs)
+            return _collect(frame)
+        elif isinstance(expr, ir.ColumnExpr):
+            # expression must be named for the projection
+            expr = expr.name('tmp').to_projection()
+            frame = self.compile(expr, params, **kwargs)
+            return _collect(frame)['tmp']
+        elif isinstance(expr, ir.ScalarExpr):
+            if expr.op().root_tables():
+                # there are associated datafusion tables so convert the expr
+                # to a selection which we can directly convert to a datafusion
+                # plan
+                expr = expr.name('tmp').to_projection()
+                frame = self.compile(expr, params, **kwargs)
+            else:
+                # doesn't have any tables associated so create a plan from a
+                # dummy datafusion table
+                compiled = self.compile(expr, params, **kwargs)
+                frame = self.context.empty_table().select(compiled)
+            return _collect(frame).iloc[0, 0]
+        else:
+            raise com.IbisError(
+                f"Cannot execute expression of type: {type(expr)}"
+            )
+
+    def compile(
+        self, expr: ir.Expr, params: Mapping[ir.Expr, object] = None, **kwargs
+    ):
+        """Compile `expr`.
+
+        Notes
+        -----
+        For the dask backend returns a dask graph that you can run ``.compute``
```

Comment:
Is this correct?

Reply:
bump ... I don't think dask is in play here right?

Reply:
Updated.